src/rust/engine/fs/src/lib.rs

```diff
@@ -297,6 +323,35 @@ impl PathGlob {
   }
 }
 
+impl fmt::Debug for PathGlob {
+  fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+    match self {
+      &PathGlob::Wildcard {
+        ref symbolic_path,
+        ref wildcard,
+        ..
```

Comment:
This is a great piece of advice!!! I think the `canonical_dir` should be added to the `Debug` for sure, I think I didn't add it because it was unnecessary screen space when I was `Debug`ging earlier, but I will add it now. I will remember your second comment for future reference though!